h/services/oauth_validator.py

```diff
@@ -125,7 +125,11 @@ def confirm_redirect_uri(
         if not redirect_uri:
             return True
 
-        return redirect_uri == client.authclient.redirect_uri
+        # Check that we match after a potentially templated redirect_uri
+        # has been templated out by our uri
+        return redirect_uri == render_url_template(
+            client.authclient.redirect_uri, example_url=redirect_uri
+        )
```

Comment:
Fix for bug 1, `confirm_redirect_uri()` did not work with templated redirect uris.